nibabel/gifti/gifti.py

```diff
@@ -238,19 +241,37 @@ class GiftiDataArray(xml.XmlSerializable):
     data = np.ndarray
     coordsys = GiftiCoordSystem
     meta = GiftiMetaData
+    """
 
-    def __init__(self, data=None):
+    def __init__(self, data=None,
+                 encoding="GIFTI_ENCODING_B64GZ",
+                 endian=sys.byteorder,
+                 coordsys=None,
+                 ordering="C",
+                 meta=None):
+        """
+        Returns a shell object that cannot be saved.
+        """
         self.data = data
         self.dims = []
         self.meta = GiftiMetaData()
         self.coordsys = GiftiCoordSystem()
         self.ext_fname = ''
         self.ext_offset = ''
+        self.intent = 0  # required attribute, NIFTI_INTENT_NONE
+        self.datatype = 0  # required attribute, void/none
+        self.ind_ord = 2  # python/numpy default: column major order
+        self.encoding = None
+        self.endian = None
```

Comment:
Maybe these should be:

    self.encoding = encoding
    self.endian = endian